TheJumpCloud/support
# Cut-4468-cascade-manager-remove-jc-user

## Issues
* [Cut-4468](https://jumpcloud.atlassian.net/browse/Cut-4468) - Cut-4468-cascade-manager-remove-jc-user

## What does this solve?
Adds functionality for cascading manager with Remove-JCUser
## Is there anything particularly tricky?
n/a
## How should this be tested?

`Remove-JCUser -UserId id`

- Delete a user/manager. If the user is a manager, there will be prompts to cascade the manager of the users
![image](https://github.com/user-attachments/assets/4277ed7a-b0ca-4064-9741-43c73f0f6c18)
- Delete manager1 that is managed by manager2. Instead of cascading the users manager to manager2. Enter a different user id to manage the managed users.
![image](https://github.com/user-attachments/assets/d1f634d3-7a41-463a-8b0d-d71aa2c59398)
- Delete manager1 that is managed by manager2. Cascade the manager2 to managed users
![image](https://github.com/user-attachments/assets/8c82cae4-f717-43a0-821b-2553eb4e6524)




`Remove-JCUser -UserId id -force`

- Remove a user(manager) with force param. This should set its managed users manager to null

`Remove-JCUser -UserId id -force -CascadeManager user -CascadeManagerUser testUser`

- Should throw an error. CascadeManager and force is not allowed

`Remove-JCUser -UserId id -CascadeManager User -CascadeManagerUser Username/Id`
![image](https://github.com/user-attachments/assets/5fe330f0-16b1-4890-9e4d-7444d8e666be)
![image](https://github.com/user-attachments/assets/6cfc7c8c-9fc5-4595-9daa-2b0ae7365f54)


- Remove a user (manager). Managed users manager should be the given Cascade Manager

`Remove-JCUser -UserId id -CascadeManager auto`

- Setup `manager1` that is managed by `manager2`. This command should automatically cascade the manager of the managed users  of manager1 to manager2 and remove/delete manager1

`Remove-JCUser -UserId id -CascadeManager null`

- Remove a user/manager. It's managed users should have no manager after removal.
## Screenshots
